aepsych/server/server.py

```diff
@@ -822,13 +825,30 @@ def tell(self, outcome, config, model_data=True):
 
         if model_data:
             x = self._config_to_tensor(config)
-            y = outcome
+            if self.has_named_outcomes:
+                y = {}
+                for out_name in self.outcome_names:
+                    ## According the above format, the outcomes are named as 'outcome_'+index
+                    index = int(out_name[-1])
+                    y[out_name] = outcome[index]
+            else:
+                y = outcome
             self.strat.add_data(x, y)
 
     def _configure(self, config):
         self.parnames = config._str_to_list(
             config.get("common", "parnames"), element_type=str
         )
+
+
+        if 'outcome_names' in config.items("common"):
```

Comment:
I think a more idiomatic way of doing this would be to do:
```
outcome_names = config.getlist("common", "outcome_names", element_type=str, fallback=None)
if outcome_names is not None:
   ...
```